Powerpc fixes

tfauck opened the pull request:
Hello,
I have rebase my work from current devel branch (did a pull of it) as you recommended.
Hope this is better this time.
Thx

markus-oberhumer commented:
Something is still wrong here - this commit affects 67 files!

Please try:

- first make sure your "devel" matches "upx/devel"

- git checkout powerpc-fixes
- make -C src/stub clean all
- git status # verify clean status

- git rebase devel
- gitk # only a few files should be affected

tfauck commented:
On 04/10/2016 19:58, Markus F.X.J. Oberhumer wrote:
>
> Something is still wrong here - this commit affects 67 files!
>
> Please try:
>
>  *
>
>     first make sure your "devel" matches "upx/devel"
>
>  *
>
>     git checkout powerpc-fixes
>
>   * make -C src/stub clean all
>  *
>
>     git status # verify clean status
>
>  *
>
>     git rebase devel
>
>   * gitk # only a few files should be affected
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/upx/upx/pull/11#issuecomment-251464204>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/ANQxUySlvwgPb0sMj9AgUOVYMkAjOfxPks5qwpPZgaJpZM4KNiuZ>.
>

Hello Markus

I have redo the work and of course my changes are related to the
following files:

 git diff --stat
 src/stub/src/arch/powerpc/64le/nrv2b_d.S | 20 ++++++++++++--------
 src/stub/src/arch/powerpc/64le/nrv2d_d.S | 24 ++++++++++++------------
 src/stub/src/arch/powerpc/64le/nrv2e_d.S | 21 ++++++++++-----------
 src/stub/src/powerpc-linux.elf-entry.S   | 73
+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++--------
 src/stub/src/powerpc-linux.elf-fold.S    |  3 ++-
 src/stub/src/powerpc-linux.elf-main.c    | 26 +++++++++++++++++++-------
 6 files changed, 120 insertions(+), 47 deletions(-)

but then my question is do I have to compile src/stub for powerpc which
would create files src/stub/src/*.h (expected but also in src/stub/tmp)
- May be then I just commit the source and *.h files to create the
complete patch ?

Thanks for your help, and sorry for my mistake taking part of your time.

Best regards

Thierry

-- 

__________________________________________________________________________
thf - Thierry Fauck - tfauck@free.fr>
/pubkey: 4096R/FCC181CE/
/fingerprint: 5CCF 6B82 DE4E E72A A40B B63E A153 BF4F FCC1 81CE/

tfauck commented:
On 04/10/2016 19:58, Markus F.X.J. Oberhumer wrote:
>
> Something is still wrong here - this commit affects 67 files!
>
> Please try:
>
>  *
>
>     first make sure your "devel" matches "upx/devel"
>
>  *
>
>     git checkout powerpc-fixes
>
>   * make -C src/stub clean all
>  *
>
>     git status # verify clean status
>
>  *
>
>     git rebase devel
>
>   * gitk # only a few files should be affected
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/upx/upx/pull/11#issuecomment-251464204>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/ANQxUySlvwgPb0sMj9AgUOVYMkAjOfxPks5qwpPZgaJpZM4KNiuZ>.
>
Hello

Would it mean I commit the following files or do I skip the bin.dump files ?

 git diff --stat |grep powerpc

 src/stub/powerpc-darwin.dylib-entry.h              |   84 +-
 src/stub/powerpc-darwin.macho-entry.h              |   74 +-
 src/stub/powerpc-darwin.macho-fold.h               |  215 +-
 src/stub/powerpc-linux.elf-entry.h                 |  431 +--
 src/stub/powerpc-linux.elf-fold.h                  |  252 +-
 src/stub/powerpc-linux.kernel.vmlinux.h            |   82 +-
 src/stub/powerpc64le-darwin.dylib-entry.h          | 1171 ++++----
 src/stub/powerpc64le-darwin.macho-entry.h          | 1089 ++++----
 src/stub/powerpc64le-darwin.macho-fold.h           |  504 +---
 src/stub/powerpc64le-linux.elf-entry.h             | 1147 ++++----
 src/stub/powerpc64le-linux.elf-fold.h              |  245 +-
 src/stub/powerpc64le-linux.kernel.vmlinux.h        | 1217 ++++-----
 src/stub/src/arch/powerpc/64le/nrv2b_d.S           |   20 +-
 src/stub/src/arch/powerpc/64le/nrv2d_d.S           |   24 +-
 src/stub/src/arch/powerpc/64le/nrv2e_d.S           |   21 +-
 src/stub/src/powerpc-linux.elf-entry.S             |   73 +-
 src/stub/src/powerpc-linux.elf-fold.S              |    3 +-
 src/stub/src/powerpc-linux.elf-main.c              |   26 +-
 src/stub/tmp/powerpc-linux.elf-entry.bin.dump      |    7 +-
 .../tmp/powerpc64le-darwin.dylib-entry.bin.dump    |   20 +-
 .../tmp/powerpc64le-darwin.macho-entry.bin.dump    |   20 +-
 src/stub/tmp/powerpc64le-linux.elf-entry.bin.dump  |   20 +-
 .../tmp/powerpc64le-linux.kernel.vmlinux.bin.dump  |   24 +-

tfauck committed:
POWERPC: Allocate 4K / 64K pages depending on processor type
    Powerpc: ppc64le processor - Fix nrv2x-d.S compression routines
        modified:   stub/src/arch/powerpc/64le/nrv2b_d.S
        modified:   stub/src/arch/powerpc/64le/nrv2d_d.S
        modified:   stub/src/arch/powerpc/64le/nrv2e_d.S
	modified:   stub/src/powerpc-linux.elf-entry.S
        modified:   stub/src/powerpc-linux.elf-fold.S
        modified:   stub/src/powerpc-linux.elf-main.c

	modified:   stub/powerpc-darwin.dylib-entry.h
	modified:   stub/powerpc-darwin.macho-entry.h
	modified:   stub/powerpc-darwin.macho-fold.h
	modified:   stub/powerpc-linux.elf-entry.h
	modified:   stub/powerpc-linux.elf-fold.h
	modified:   stub/powerpc-linux.kernel.vmlinux.h
	modified:   stub/powerpc64le-darwin.dylib-entry.h
	modified:   stub/powerpc64le-darwin.macho-entry.h
	modified:   stub/powerpc64le-darwin.macho-fold.h
	modified:   stub/powerpc64le-linux.elf-entry.h
	modified:   stub/powerpc64le-linux.elf-fold.h
	modified:   stub/powerpc64le-linux.kernel.vmlinux.h
	modified:   stub/scripts/xstrip.py
	modified:   stub/src/arch/powerpc/64le/nrv2b_d.S
	modified:   stub/src/arch/powerpc/64le/nrv2d_d.S
	modified:   stub/src/arch/powerpc/64le/nrv2e_d.S
	modified:   stub/src/powerpc-linux.elf-entry.S
	modified:   stub/src/powerpc-linux.elf-fold.S
	modified:   stub/src/powerpc-linux.elf-main.c

tfauck commented:
I have updated my branch properly (without the dump files) so with only the .S and .h files generated meaning 18 files are being modified ...and apparently travis seems much more happy. May be closing that request and opening a new one would be more efficient - what do you prefer ? Thx

tfauck commented:
On 04/10/2016 19:58, Markus F.X.J. Oberhumer wrote:
>
> Something is still wrong here - this commit affects 67 files!
>
> Please try:
>
>  *
>
>     first make sure your "devel" matches "upx/devel"
>
>  *
>
>     git checkout powerpc-fixes
>
>   * make -C src/stub clean all
>  *
>
>     git status # verify clean status
>
>  *
>
>     git rebase devel
>
>   * gitk # only a few files should be affected
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/upx/upx/pull/11#issuecomment-251464204>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/ANQxUySlvwgPb0sMj9AgUOVYMkAjOfxPks5qwpPZgaJpZM4KNiuZ>.
>
OK - Found my problem - when I do clean all it deletes everything in
src/stub/tmp and because I am only compiling

powerpc stuff , some files are found to be removed explaining why I get
133 files changed instead of the 18 files I expect.

Going to redo that differently - please close the bad  pull request and
I hope to have it clean now.

Thx

markus-oberhumer commented:
Hi Thierry,

On 2016-10-11 14:14, tfauck wrote:
> Hello
> 
> Would it mean I commit the following files or do I skip the bin.dump files ?
> 
>  git diff --stat |grep powerpc
> 
>  src/stub/powerpc-darwin.dylib-entry.h              |   84 +-
>  src/stub/powerpc-darwin.macho-entry.h              |   74 +-
>  src/stub/powerpc-darwin.macho-fold.h               |  215 +-
>  src/stub/powerpc-linux.elf-entry.h                 |  431 +--
>  src/stub/powerpc-linux.elf-fold.h                  |  252 +-
>  src/stub/powerpc-linux.kernel.vmlinux.h            |   82 +-
>  src/stub/powerpc64le-darwin.dylib-entry.h          | 1171 ++++----
>  src/stub/powerpc64le-darwin.macho-entry.h          | 1089 ++++----
>  src/stub/powerpc64le-darwin.macho-fold.h           |  504 +---
>  src/stub/powerpc64le-linux.elf-entry.h             | 1147 ++++----
>  src/stub/powerpc64le-linux.elf-fold.h              |  245 +-
>  src/stub/powerpc64le-linux.kernel.vmlinux.h        | 1217 ++++-----
>  src/stub/src/arch/powerpc/64le/nrv2b_d.S           |   20 +-
>  src/stub/src/arch/powerpc/64le/nrv2d_d.S           |   24 +-
>  src/stub/src/arch/powerpc/64le/nrv2e_d.S           |   21 +-
>  src/stub/src/powerpc-linux.elf-entry.S             |   73 +-
>  src/stub/src/powerpc-linux.elf-fold.S              |    3 +-
>  src/stub/src/powerpc-linux.elf-main.c              |   26 +-
>  src/stub/tmp/powerpc-linux.elf-entry.bin.dump      |    7 +-
>  .../tmp/powerpc64le-darwin.dylib-entry.bin.dump    |   20 +-
>  .../tmp/powerpc64le-darwin.macho-entry.bin.dump    |   20 +-
>  src/stub/tmp/powerpc64le-linux.elf-entry.bin.dump  |   20 +-
>  .../tmp/powerpc64le-linux.kernel.vmlinux.bin.dump  |   24 +-

yes, this is the expected diff - all other files should be left untouched.

BTW, you can also "git clone https://github.com/upx/upx-testsuite.git"
and try running "make run-testsuite" on your local machine - please
see src/Makefile.

And as you made some changes to the powerpc stubs you may have to
update the sha256sums - please see "recreate_expected_sha256sums"
in upx/.github/travis_testsuite_1.sh .

Thanks,
Markus



> 
> 
> 


-- 
Markus Oberhumer, <markus@oberhumer.com>, http://www.oberhumer.com/
